File: src/HotelRestaurant.Application/Services/Implementations/HotelService.cs
using HotelRestaurant.Application.DTOs;
using HotelRestaurant.Application.Services.Interfaces;
using HotelRestaurant.Core.Entities;
using HotelRestaurant.Core.Interfaces;

namespace HotelRestaurant.Application.Services.Implementations
{
    public class HotelService : IHotelService
    {
        private readonly IUnitOfWork _unitOfWork;

        public HotelService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }

        public async Task<IEnumerable<HotelLookupDto>> GetLookupAsync()
        {
            var hotels = await _unitOfWork.Hotels.GetAllAsync();
            
            return hotels
                .Where(h => !h.IsDeleted)
                .Select(h => new HotelLookupDto
                {
                    Id = h.Id,
                    Name = h.Name
                });
        }

        public async Task<IEnumerable<HotelDto>> GetAllAsync()
        {
            var hotels = await _unitOfWork.Hotels.GetAllAsync();
            return hotels
                .Where(h => !h.IsDeleted)
                .Select(MapToDto);
        }

        public async Task<HotelDto?> GetByIdAsync(int id)
        {
            var hotel = await _unitOfWork.Hotels.GetByIdAsync(id);
            if (hotel == null || hotel.IsDeleted) return null;
            
            return MapToDto(hotel);
        }

        public async Task<HotelDto> CreateAsync(CreateHotelDto dto)
        {
            var hotel = new HotelRestaurant.Core.Entities.Hotel
            {
                Name = dto.Name,
                Address = dto.Address,
                City = dto.City,
                Country = dto.Country,
                Phone = dto.Phone,
                Email = dto.Email,
                //Rating = dto.Rating
            };

            await _unitOfWork.Hotels.AddAsync(hotel);
            await _unitOfWork.SaveChangesAsync();

            return MapToDto(hotel);
        }

        private static HotelDto MapToDto(HotelRestaurant.Core.Entities.Hotel hotel) => new()
        {
            Id = hotel.Id,
            Name = hotel.Name,
            Address = hotel.Address,
            City = hotel.City,
            Country = hotel.Country,
            Phone = hotel.Phone,
            Email = hotel.Email,
            //Rating = hotel.Rating
        };
    }
}

File: client-app/src/app/services/hotel.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { apiBaseUrl } from '../app.config';


export interface HotelLookupDto {
  id: number;
  name: string;
}

export interface HotelDto {
  id: number;
  name: string;
  address: string;
  city: string;
  country: string;
  phone: string;
  email: string;
  rating: number;
}

export interface CreateHotelDto {
  name: string;
  address: string;
  city: string;
  country: string;
  phone: string;
  email: string;
  rating: number;
}

@Injectable({
  providedIn: 'root'
})
export class HotelService {
  private readonly apiUrl = `${apiBaseUrl}/api/hotels`;

  constructor(private readonly http: HttpClient) {}

  getLookup(): Observable<HotelLookupDto[]> {
    return this.http.get<HotelLookupDto[]>(`${this.apiUrl}/lookup`);
  }

  getAll(): Observable<HotelDto[]> {
    return this.http.get<HotelDto[]>(this.apiUrl);
  }

  getById(id: number): Observable<HotelDto> {
    return this.http.get<HotelDto>(`${this.apiUrl}/${id}`);
  }

  create(dto: CreateHotelDto): Observable<HotelDto> {
    return this.http.post<HotelDto>(this.apiUrl, dto);
  }
}